### DISCUSSION TITLE
[Support]: Cannot export videos longer than 20min

### DISCUSSION THREAD

**Discussion by YenLegion** [blakeblackshear/frigate#11861] [General Support] [answered]
REDACTED

**Comment by NickM-27** [comment]
there seems to be some confusion about a few things here:
1. just adding a second stream `      - ffmpeg:southwest#video=h264#audio=copy#audio=opus` will do absolutely nothing. The first stream in the list will satisfy the request and the second line will not be used for video.
2. There is no GPU usage because exporting video does not use the GPU because there is no encoding going on, it is just mp4 packaging (unless you do a time-lapse export). This just seems like a system issue

**Reply by YenLegion** [reply]
All I need on the second (sub)stream is "- ffmpeg:southwest" for detect and audio roles? 

I don't know if I explained well, but I had just been copying from the main stream for recording. Yesterday I switched to reencode to h264 on the main (recording) stream, just to see if it would help. Exporting right now actually to test. 

Thank you.

**Reply by NickM-27** [reply, marked as answer]
> Yesterday I switched to reencode to h264 on the main (recording) stream, just to see if it would help. Exporting right now actually to test.

the config you have here WILL NOT re-encode anything

I think you are misunderstanding how this works. Adding a second line does not change the original stream, it adds to it.

For example if you have 

```
go2rtc:
  streams:

    southwest:
      - rtsp://REDACTED/cam/realmonitor?channel=1&subtype=0&authbasic=REDACTED
      - ffmpeg:southwest#video=h264#audio=copy#audio=opus
```

go2rtc will check what frigate is asking for (in this case any video and aac audio). If that first stream provides it then go2rtc stops there, otherwise it moves on to the second line. 

So adding a second line is only useful when defining additional things, in this case you might want opus audio so webrtc will have audio

**Reply by YenLegion** [reply]
Ok my mistake, I need to spend some more time with the documentation. Thank you for the pointers.

**Reply by NickM-27** [reply]
logs will definitely be helpful in the case that it fails. You can also check the nginx logs to see if the VOD files are still being pulled down which is a sign that it is still doing work.

**Reply by YenLegion** [reply]
I think it is a system issue like you mentioned. Just tried to export an hour and the entire VM eventually locked up. Planning to ditch ESXi here soon.
